Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -177,11 +190,11 @@ void UPSSpotComponent::RefreshAmountOfUnlockedSkins(bool bApplySkin)
 
 	UMySkeletalMeshComponent& SpotMeshComponent = GetMeshChecked();
 	const int32 UnlockedSkinsAmount = UPSWorldSubsystem::Get().GetCurrentSaveToDiskRowByName().UnlockedSkinsAmount;
-
-	for (int32 Index = 0; Index <= UnlockedSkinsAmount; Index++)
+	const int32 CurrentSkinIndex = SpotMeshComponent.GetAppliedSkinIndex();
+	for (int32 Index = CurrentSkinIndex; Index <= UnlockedSkinsAmount; Index++)
 	{
 		SpotMeshComponent.SetSkinAvailable(true, Index);
-		if (bApplySkin)
+		if (bApplySkin && PreviousUnlockedAmountOfSkins != UnlockedSkinsAmount)
```

Comment:
I would try to avoid adding new property, as always for reason of its supporting cost. Instead, please try different approaches.

For instance, dont call `PlayerCharacter->SetCustomPlayerMeshData` on playground character if current state is not menu. But once it's menu, force applying on playground character.

If this idea does not work for any reason, please investigate what else can be done without introducing PreviousUnlockedAmountOfSkins

Reply:
done as part of: https://github.com/moinm3uw/ProgressionSystem/pull/67/commits/656605cc9c6bedef2bf45b8fb59a7f93e83ab186